web/src/util/serialization/apd.js

```diff
@@ -41,37 +41,33 @@ export const fromAPI = apdAPI => {
         statePersonnel,
         ...activity
       }) => ({
-        key: generateKey(),
-
         ...activity,
 
+        key: generateKey(),
+
         // These properties need some massaging into reducer state
         contractorResources: contractorResources.map(c => ({
-          key: generateKey(),
-          ...c
+          ...c,
+          key: generateKey()
```

Comment:
Invert these so the local `key` will always override whatever we get from the server. It shouldn't send a `key` property at all, but this is a safeguard.